packages/shared/src/hooks/usePostById.ts

```diff
@@ -38,36 +37,21 @@ export const updatePostCache = (
     },
   }));
 
-const usePostById = ({
-  id,
-  isFetchingNextPage,
-  options = {},
-}: UsePostByIdProps): UsePostById => {
+const usePostById = ({ id, options = {} }: UsePostByIdProps): UsePostById => {
   const { tokenRefreshed } = useAuthContext();
   const key = getPostByIdKey(id);
-  const {
-    data: postById,
-    isLoading,
-    isFetched,
-  } = useQuery<PostData>(
+  const query = useQuery<PostData>(
     key,
     () => request(graphqlUrl, POST_BY_ID_QUERY, { id }),
     {
       ...options,
       enabled: !!id && tokenRefreshed,
     },
   );
-
+  const { data: postById } = query;
   const post = postById || (options?.initialData as PostData);
 
-  return useMemo(
-    () => ({
-      post: post?.post,
-      isFetched,
-      isLoading: isLoading || isFetchingNextPage,
-    }),
-    [id, postById, isLoading, options, isFetched, isFetchingNextPage],
-  );
+  return useMemo(() => ({ post: post?.post, query }), [query]);
```

Comment:
Think you will always need `id` here to reset memo no?
Not sure query alone will fetch that.

Reply:
The id is part of the query key, so when its key changes, it re-fires which will then update the `query`, and since the `query` gets updated, this gets reset. But I can definitely add the `id` for dependency to avoid any weird edge cases.